keyboards/planck/keymaps/stuartfong1/keymap.c

```diff
@@ -0,0 +1,312 @@
+     * |  F5  |  F6  |  F7  |  F8  |PrtSc4|      |      |      |  F17 |  F18 |  F19 |  F20 |
+     * |------+------+------+------+------+------+------+------+------+------+------+------|
+     * |  F9  |  F10 |  F11 |  F12 |PrtSc3|      |      |      |  F21 |  F22 |  F23 |  F24 |
+     * |------+------+------+------+------+------+------+------+------+------+------+------|
+     * |      |      |      |      |      |      |      |      |      |      |      |      |
+     * `-----------------------------------------------------------------------------------'
+     */
+    [_FUNCTION] = LAYOUT_planck_grid(
+        KC_F1,   KC_F2,   KC_F3,   KC_F4,   PRTSC5,  XXXXXXX, XXXXXXX, XXXXXXX, KC_F13,  KC_F14,  KC_F15,  KC_F16,
+        KC_F5,   KC_F6,   KC_F7,   KC_F8,   PRTSC4,  XXXXXXX, XXXXXXX, XXXXXXX, KC_F17,  KC_F18,  KC_F19,  KC_F20,
+        KC_F9,   KC_F10,  KC_F11,  KC_F12,  PRTSC3,  XXXXXXX, XXXXXXX, XXXXXXX, KC_F21,  KC_F22,  KC_F23,  KC_F24,
+        XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX
+    ),
+
+    /* Adjust (Lower + Raise)
+     *                      v------------------------RGB CONTROL--------------------v
+     * ,-----------------------------------------------------------------------------------.
+     * |      | Reset|Debug | RGB  |RGBMOD| HUE+ | HUE- | SAT+ | SAT- |BRGTH+|BRGTH-|      |
+     * |------+------+------+------+------+------+------+------+------+------+------+------|
+     * |      |      |MUSmod|Aud on|Audoff|AGnorm|AGswap|      |Plover|      |      |      |
+     * |------+------+------+------+------+------+------+------+------+------+------+------|
+     * |      |Voice-|Voice+|Mus on|Musoff|MIDIon|MIDIof|TermOn|TermOf|      |      |      |
+     * |------+------+------+------+------+------+------+------+------+------+------+------|
+     * |      |      |      |      |      |      |      |      |      |      |      |      |
+     * `-----------------------------------------------------------------------------------'
+     */
+    [_ADJUST] = LAYOUT_planck_grid(
+        XXXXXXX, RESET,   DEBUG,   RGB_TOG, RGB_MOD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD,  RGB_VAI, RGB_VAD, XXXXXXX, 
+        XXXXXXX, XXXXXXX, MU_MOD,  AU_ON,   AU_OFF,  AG_NORM, AG_SWAP, XXXXXXX, PLOVER,   XXXXXXX, XXXXXXX, XXXXXXX, 
+        XXXXXXX, MUV_DE,  MUV_IN,  MU_ON,   MU_OFF,  MI_ON,   MI_OFF,  TERM_ON, TERM_OFF, XXXXXXX, XXXXXXX, XXXXXXX, 
+        XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,  XXXXXXX, XXXXXXX, XXXXXXX
+    )
+};
+
+#ifdef AUDIO_ENABLE
+    float plover_song[][2]     = SONG(PLOVER_SOUND);
+    float plover_gb_song[][2]  = SONG(PLOVER_GOODBYE_SOUND);
+#endif
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+    switch (biton32(state)) {
+        case _PLOVER:
+            autoshift_disable();
+            break;
+        default:
+            autoshift_enable();
+            break;
+    }
```

Comment:
```suggestion
    bool is_plover_on = layer_state_cmp(state, _PLOVER);
    bool static was_plover_on = false, autoshift_on;
    if (is_plover_on != was_plover_on) {
        if (is_plover_on) {
            autoshift_on = get_autoshift_state();
            autoshift_disable();
        } else {
            if (autoshift_on) {
                autoshift_enable();
                autoshift_on = false;
            }
        }
        was_plover_on = is_plover_on;
     }
```

This is more complex, but the gist of it is, that it tracks the previous state, and if plover is turned on, it gets the autoshift state, saves it, and reverts back to that state when its turned off. 

This way, you can turn off autoshift, without having to worry about it being forced on, if plover is enabled.